### DISCUSSION TITLE
Need access to `personal-files` interface

### DISCUSSION THREAD

**Discussion by scurvydoggo** [jgneff/strictly-maven#2] [General]
Hello, I am using this snap since Pop OS (ubuntu) does not have the latest `mvn` via the repository.

In my use case, maven is trying to access a hidden configuration file in my home directory. From reading the docs, hidden files are not exposed via the `home` interface, and instead the `personal-files` interface should be used.

Here is my error:

```
➜  strictly-maven validate -Dset.changelist -Dignore.dirt
[ERROR] Git operations failed: /home/me/.config/jgit/config (Permission denied) -> [Help 1
```

The command itself is validating a Jenkins plugin project which leverages some maven plugins to perform CD automation.

**Comment by jgneff** [comment]
Below are a few things you might try to work around the problem:

* Set the `GIT_CONFIG_NOSYSTEM` environment variable before running the command. That lets JGit [avoid reading the file][1], and might allow the command to continue without errors.

  ```console
  $ export GIT_CONFIG_NOSYSTEM=true
  $ strictly-maven validate -Dset.changelist -Dignore.dirt
  ```

* Build your fork of this repository with one of the commands shown below. The first command builds it locally in a container or VM on your system, while the second builds it on the Launchpad Build Farm (and requires a Launchpad account).

  ```console
  $ snapcraft --debug
  $ snapcraft remote-build
  ```

  Once built, you can install it and connect the `personal-files` interface with the commands:

  ````
  $ sudo snap install strictly-maven_3.9.3_multi.snap --dangerous
  $ sudo snap connect strictly-maven:dot-config-jgit
  ````

* If the security aspects of the Strictly Maven package are not your main priority, download and install the [official Maven package][2] directly from Apache.

You don't need to publish a Snap package if you just want a private copy for your own system. I find it's a great way to build and install the latest version of the software I need when it's not available from my distribution and I don't want to download, verify, and install the software from the vendor's website.

[1]: https://git.eclipse.org/c/jgit/jgit.git/tree/org.eclipse.jgit/src/org/eclipse/jgit/util/SystemReader.java#n103
[2]: https://maven.apache.org/download.cgi

**Reply by scurvydoggo** [reply]
Ok thanks for the clarification. I will try with the official package - no security requirements on this.
